File: applets/src/org/rr0/ufoathome/view/gui/MessageEvent.java
package org.rr0.ufoathome.view.gui;

import java.util.EventObject;

/**
 * @author Jerôme Beau
 * @version 0.3
 */
public class MessageEvent extends EventObject {
    private String message;
    private MessageEditable editable;

    public MessageEvent(Object source, String message, MessageEditable editable) {
        super(source);
        this.message = message;
        this.editable = editable;
    }

    public MessageEditable getEditable() {
        return editable;
    }

    public String getText() {
        return message;
    }
}

File: applets/src/org/rr0/ufoathome/view/gui/MessageEvent.ts
import {EventObject} from "./EventObject";
import {MessageEditable} from "./MessageEditable";

export class MessageEvent extends EventObject {
  private message: string;
  private editable: MessageEditable;

  constructor(source: Object, message: String, editable: MessageEditable) {
    super(source);
    this.message = message;
    this.editable = editable;
  }

  public getEditable(): MessageEditable {
    return this.editable;
  }

  public getText(): string {
    return this.message;
  }
}